src/components/Select/hooks.js

```diff
@@ -0,0 +1,195 @@
+import { useEffect, useState } from "react";
+import { useClickOutside } from "hooks";
+import { blankValue, DOWN, findExact, UP } from "./helpers";
+import { useTransition } from "react-spring";
+
+export function useSelect(
+  value,
+  disabled,
+  getOptionLabel,
+  getOptionValue,
+  autoFocus,
+  onChange,
+  inputValue,
+  onInputChange,
+  searchable
+) {
+  const [currentOptions, setCurrentOptions] = useState([]);
+  const [menuOpened, setMenuOpened] = useState(false);
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState(0);
+  const [showError, setShowError] = useState(false);
+
+  useEffect(() => {
+    if (disabled) {
+      setMenuOpened(false);
+      if (searchable && onInputChange && inputValue) onInputChange("");
+      return;
+    }
+    if (autoFocus) setMenuOpened(true);
+  }, [disabled, autoFocus, searchable, inputValue, onInputChange]);
+
+  useEffect(() => {
+    if (searchable && inputValue && !showError)
+      setMenuOpened(currentOptions.length > 0);
+  }, [searchable, inputValue, currentOptions, showError]);
+
+  const resetMenu = (focusedIndex = 0) => {
+    setFocusedOptionIndex(focusedIndex);
+    setMenuOpened(false);
+    if (searchable && onInputChange && inputValue) onInputChange("");
+  };
+
+  const removeSelectedOptionFilter = (option) =>
+    value.filter(
+      (selectedOption) =>
+        getOptionLabel(selectedOption) !== getOptionLabel(option)
+    );
+
+  const setOption = (option, knownIndex) => {
+    if (disabled) return;
+    const index =
+      knownIndex ||
+      findExact(currentOptions, getOptionLabel, getOptionValue, option);
+    let newOptions;
+    if (Array.isArray(value)) {
+      newOptions = removeSelectedOptionFilter(option);
+      if (value.length === newOptions.length) {
+        if (!value.length) {
+          newOptions = [option];
+        } else {
+          newOptions = [...value, option];
```

Comment:
```suggestion
        newOptions = [...value, option];
```
you are already checking if `value` is an Array.